### ISSUE TITLE
Share links with fragment identifiers broken for Twitter/X

### ISSUE THREAD

**Issue by TimWeberRE100** [TerriaJS/terriajs#7291]
Unfortunately it is not possible to post links to twitter/X that contain fragment identifiers (#) because of the way that hashtags work on their website. This means that share links (including shortened share links based on Github Gist) cannot be directly shared to Twitter/X (one workaround is to just use another URL shortening service). Is it feasible to include share links that use a query parameter (?) instead of the #?

**Comment by na9da** [comment]
Hi @TimWeberRE100 

Not sure if something changed recently on X, but here's an example of a tweet with `#share` links

https://x.com/SatelliteSci/status/1537308160930639872

One other option is to use the story url feature which is meant to create vanity urls like `/story/coolstory`.

To do this you need to set the following option in your config.json
```json
"storyRouteUrlPrefix": "https://my-map.com/share/s-"
```

after that you can access your share URLs like `https://my-map.com/story/<short-id>`

basically, to resolve `/story/short-id`, Terria takes the `short-id` and appends it to the `storyRouteUrlPrefix` and fetches the share JSON from there.

So an alternate URL for the link in that tweet would be this:
https://maps.dea.ga.gov.au/story/w256tDNTchZcRsoyM2CerdxnDO1
